unit/src/MFEMEssentialBCTest.C

```diff
@@ -0,0 +1,380 @@
+#include "mfem-common.hpp"
+#include "MFEMObjectUnitTest.h"
+#include "MFEMScalarDirichletBC.h"
+#include "MFEMScalarFunctionDirichletBC.h"
+#include "MFEMVectorDirichletBC.h"
+#include "MFEMVectorFunctionDirichletBC.h"
+#include "MFEMVectorNormalDirichletBC.h"
+#include "MFEMVectorFunctionNormalDirichletBC.h"
+#include "MFEMVectorTangentialDirichletBC.h"
+#include "MFEMVectorFunctionTangentialDirichletBC.h"
+
+class MFEMEssentialBCTest : public MFEMObjectUnitTest
+{
+public:
+  mfem::common::H1_FESpace _scalar_fes;
+  mfem::common::H1_FESpace _vector_h1_fes;
+  mfem::common::ND_FESpace _vector_hcurl_fes;
+  mfem::common::RT_FESpace _vector_hdiv_fes;
+  mfem::GridFunction _scalar_gridfunc, _vector_h1_gridfunc, _vector_hcurl_gridfunc,
+      _vector_hdiv_gridfunc;
+  mfem::ConstantCoefficient _scalar_zero;
+  mfem::VectorConstantCoefficient _vector_zero;
+
+  MFEMEssentialBCTest()
+    : MFEMObjectUnitTest("PlatypusApp"),
+      _scalar_fes(_mfem_mesh_ptr->getMFEMParMeshPtr().get(), 1, 3),
+      _vector_h1_fes(
+          _mfem_mesh_ptr->getMFEMParMeshPtr().get(), 1, 3, mfem::BasisType::GaussLobatto, 3),
+      _vector_hcurl_fes(_mfem_mesh_ptr->getMFEMParMeshPtr().get(), 2, 3, 3),
+      _vector_hdiv_fes(_mfem_mesh_ptr->getMFEMParMeshPtr().get(), 2, 3, 3),
```

Comment:
This is currently causing the vector tests to fail for me - `vdim` should instead be equal to 1 for the vector FE types (scalar coefficients but vector basis functions, as opposed to the H1 vector case)

```suggestion
      _vector_hcurl_fes(_mfem_mesh_ptr->getMFEMParMeshPtr().get(), 2, 3),
      _vector_hdiv_fes(_mfem_mesh_ptr->getMFEMParMeshPtr().get(), 2, 3),
```

Reply:
Funny, given it worked on CI and on my computer.